node/libs/storage/src/testonly/mod.rs

```diff
@@ -29,17 +32,66 @@ impl Distribution<ReplicaState> for Standard {
     }
 }
 
+/// Test-only memory storage for blocks and batches.
+pub struct TestMemoryStorage {
+    /// In-memory block store with its runner.
```

Comment:
There is no point in exposing multiple runners - wrap them into a single runner which just spawns those 2 inside.